Framework/PythonInterface/plugins/algorithms/IntegratePeaksSkew.py

```diff
@@ -0,0 +1,714 @@
+                # window within ncols_edge of detector edge on LHS of the bank with no adjacent tube in another
+                col_edges = dcol <= ncols_edge - col
+            elif isRHS and dcol[0, -1] > ncols - col - ncols_edge:
+                # window within ncols_edge of detector edge on RHS of the bank with no adjacent tube in another
+                col_edges = dcol > ncols - col - ncols_edge
+        row_edges = np.logical_or(drow <= ndet_per_spec * nrows_edge - row,
+                                  drow > nrows - row - ndet_per_spec * nrows_edge)
+        detector_edges = np.logical_or(row_edges, col_edges)
+        # row_peak, icol_peak
+        irow_peak = np.where(drow[:, 0] == 0)[0][0]
+        icol_peak = np.where(dcol[0, :] == 0)[0][0]
+        return detids, detector_edges, irow_peak, icol_peak
+
+    def _get_detid_array_rect_detector(self, bank, detid, row, col, drows, dcols, nrows_edge, ncols_edge):
+        col_step, row_step = bank.idstep(), bank.idstepbyrow()  # step in detID along col and row
+        if bank.idfillbyfirst_y():
+            col_step, row_step = row_step, col_step
+        # need to adjust range depending on whether above min/max row/col
+        drow_vec = np.arange(max(0, row - drows), min(row + drows + 1, bank.xpixels())) - row
+        dcol_vec = np.arange(max(0, col - dcols), min(col + dcols + 1, bank.ypixels())) - col
+        dcol, drow = np.meshgrid(dcol_vec, drow_vec)
+        detids = detid + dcol * col_step + drow * row_step
+        # create bool mask for detector edges
+        detector_edges = np.logical_or.reduce((drow <= -row + nrows_edge - 1, dcol <= -col + ncols_edge - 1,
+                                               drow >= bank.xpixels() - nrows_edge - row,
+                                               dcol >= bank.ypixels() - ncols_edge - col))
+        # get indices of peak centre
+        irow_peak = np.where(drow_vec == 0)[0][0]
+        icol_peak = np.where(dcol_vec == 0)[0][0]
+        return detids, detector_edges, irow_peak, icol_peak
+
+    def get_peak_region_array(self, peak, detid, bank_name, drows, dcols, nrows_edge, ncols_edge):
+        """
+        :param peak: peak object
+        :param detid: detector id of peak (from peak table)
+        :param bank_name: bank name on which detector resides (from peak table)
+        :param dpixel: width of detector window in pixels (along row and columns)
+        :return signal: 3D numpy array containing signal/intensities (row x cols x bins)
+        :return errors: 3D numpy array containing errors (row x cols x bins)
+        :return irow_peak: index of peak position along first dim (row)
+        :return icol_peak: index of peak position along second dim (col)
+        :return ispec: spectrum index corresponding to detid of peak
+        :return edges: bool mask True for pixels on the edge of a detector bank/panel
+        """
+        bank = self.inst.getComponentByName(bank_name)
```

Comment:
I think it's worth checking bank isn't `None` here (or inside get_detid_array). If the peaks workspace doesn't line up with the input workspace this might happen. Again this comes from running the test script on the PR with an input workspace instead of SXD23767

Reply:
I will also add a check in the input validation that the instrument of the peak workspace is the same as the raw workspace!

Reply:
OK that sounds good. Hadn't realised the peaks workspaces had an instrument

Reply:
As in a previous comment I have added input validation to make sure the pea and data workspace have the same instrument - if that's the case then the bank name must correspond to a instrument component.